lib/aws-s3-v2-wrapper.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+const {S3Client, GetObjectCommand} = require('@aws-sdk/client-s3')
+const {object} = require('@logdna/stdlib')
+
+const s3 = new S3Client({})
+
+module.exports = exports = {
+  getObject
+}
+
+async function getObject(params) {
+  let data, err
+  try {
+    const command = new GetObjectCommand(params)
+    data = await s3.send(command)
+  } catch (error) {
+    err = new Error('Error in Getting the S3 Object')
+    err.meta = {error, params}
+    throw err
+  }
+
+  if (!object.has(data, 'Body')) {
+    err = new Error('Corrupted data returned from the object')
+    err.meta = {params}
+    throw err
+  }
+
+  const s3ResponseStream = data.Body
```

Comment:
Body works much differently than in v2.  So I moved most of this logic out to it's own function.